src/unit_tests.rs

```diff
@@ -548,3 +399,132 @@ fn adding_slots_to_each_other_works() {
         slot_a + slot_h
     );
 }
+
+fn get_test_tasks_2() -> Vec<Task> {
+    vec![
+        Task {
+            id: 1,
+            goal_id: 2,
+            title: "clean car".to_string(),
+            duration: 2,
+            status: TaskStatus::UNSCHEDULED,
+            flexibility: 0,
```

Comment:
Guess it is ok to initialize this with 0 if flex gets recalculated before any action happens.

Reply:
Yes it does. The alternative is to use an `Option` but this makes it unnecessarily verbose.

Reply:
If we include setting the correct flexibility after the Slot assignment/generation at Task initialization time, we can make it responsible for always having the correct flexibility at hand.

It can then be done on initialization + after every edit to its slots. Accessing it would then just be a getter to the internal value (like a cache) - not a new calculation every time.